packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -480,21 +478,32 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
 
   // Response for component islands
   if (isRenderingIsland && islandContext) {
-    const islandHead: NuxtIslandResponse['head'] = {
-      link: [],
-      style: [],
-    }
-    for (const tag of await head.resolveTags()) {
-      if (tag.tag === 'link') {
-        islandHead.link.push({ key: 'island-link-' + hash(tag.props), ...tag.props })
-      } else if (tag.tag === 'style' && tag.innerHTML) {
-        islandHead.style.push({ key: 'island-style-' + hash(tag.innerHTML), innerHTML: tag.innerHTML })
-      }
-    }
+    // re-push inlined styles - head is reset by the island renderer
+    head.push({ style: inlinedStyles })
     const islandResponse: NuxtIslandResponse = {
       id: islandContext.id,
-      head: islandHead,
-      html: getServerComponentHTML(htmlContext.body),
+      head: head.headEntries().filter((h) => {
+        if (h.resolvedInput) {
+          const resolvedInput = toValue(h.resolvedInput)
+          for (const key in resolvedInput) {
+            const input = resolvedInput[key as keyof MergeHead]
+            if (Array.isArray(input)) {
+              if (toValue(input).length === 0) {
+                delete resolvedInput[key as keyof MergeHead]
+              } else {
+                // @ts-expect-error todo fix this
+                input.forEach((i) => { if (typeof i === 'object') { i.key = 'island-key-' + hash(i) } })
```

Comment:
what's this for?

Reply:
This was implemented for tag deduplication I think, @danielroe can we drop the `key` property ?

Reply:
see here https://github.com/nuxt/nuxt/blob/128edc76fc0879119525fccee63f504657b36d63/packages/nuxt/src/core/runtime/nitro/renderer.ts#L470-L477

Reply:
Link and style tags can't dedupe themselves so I can see why this exists (if someone is rendering multiple of the same component).